src/compiler/crystal/codegen/fun.cr

```diff
@@ -406,7 +424,8 @@ class Crystal::CodeGenVisitor
         def_var = def_vars.try &.[var.name]?
         next if def_var && !def_var.closured?
 
-        self.context.vars[var.name] = LLVMVar.new(gep(closure_ptr, 0, i, var.name), var.type)
+        ptr = gep(closure_ptr, 0, i, var.name)
+        self.context.vars[var.name] = LLVMVar.new(ptr, var.type)
```

Comment:
Is this change necessary?

Reply:
I did roll it back in one place but missed in other. I will roll it back.